File: src/main/java/cc/aoeiuv020/panovel/server/dal/model/MobResponse.kt
package cc.aoeiuv020.panovel.server.dal.model

import cc.aoeiuv020.panovel.server.common.ErrorCode

/**
 *
 * Created by AoEiuV020 on 2018.04.02-11:21:33.
 */
class MobResponse(
        var code: Int = ErrorCode.UNKNOWN_ERROR.code,
        var data: Any? = null
) {
    fun setError(error: ErrorCode) {
        code = error.code
    }
}

File: model/MobResponse.php
<?php
/**
 * Created by AoEiuV020 on 2018.06.22-08:30:05.
 */

class MobResponse
{
    public $code = -1;
    public $data = "{}";

    public function serverError($message)
    {
        $this->code = 500;
        $this->data = json_encode($message);
    }

    public function illegalRequest($message)
    {
        $this->code = 400;
        $this->data = json_encode($message);
    }

    public function success($data)
    {
        $this->code = 200;
        $this->data = json_encode($data);
    }
}